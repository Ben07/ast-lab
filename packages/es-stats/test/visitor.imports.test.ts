import fs from 'fs';
import path from 'path';
import { parse } from '@babel/parser';
import traverse from '@babel/traverse';
import { Import } from 'ast-lab-types';
import createVisitor from '../src/visitors/imports';

const dir = path.resolve(__dirname, '__fixtures__/imports');
const files = fs.readdirSync(dir);
describe('import visitors', () => {
  files.forEach(file => {
    test(file, () => {
      let res = [] as Import[];
      const code = fs.readFileSync(path.resolve(dir, file), 'utf-8');
      traverse(
        // @ts-ignore
        parse(code, { 
          sourceType: 'module',
          plugins: ['dynamicImport']
        }),
        createVisitor(res)
      );
      expect(res).toMatchSnapshot();
    });
  });
});
