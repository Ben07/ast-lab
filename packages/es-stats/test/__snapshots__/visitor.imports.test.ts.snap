// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`import visitors exports/exportAllFrom.js 1`] = `Array []`;

exports[`import visitors exports/exportDefault.js 1`] = `Array []`;

exports[`import visitors exports/exportDefaultFuncAlias.js 1`] = `Array []`;

exports[`import visitors exports/exportDefaultValueAlias.js 1`] = `Array []`;

exports[`import visitors exports/exportFunc.js 1`] = `Array []`;

exports[`import visitors exports/exportFuncAlias.js 1`] = `Array []`;

exports[`import visitors exports/exportNamed.js 1`] = `Array []`;

exports[`import visitors exports/exportNamedFrom.js 1`] = `Array []`;

exports[`import visitors exports/noExports.js 1`] = `Array []`;

exports[`import visitors imports/dynamicImport-asyncawait.js 1`] = `
Array [
  Object {
    "alias": "dynamicImport",
    "loc": SourceLocation {
      "end": Position {
        "column": 44,
        "line": 2,
      },
      "start": Position {
        "column": 30,
        "line": 2,
      },
    },
    "name": "dynamicImport",
    "source": "./b",
  },
]
`;

exports[`import visitors imports/dynamicImport-then.js 1`] = `
Array [
  Object {
    "alias": "./b",
    "loc": SourceLocation {
      "end": Position {
        "column": 16,
        "line": 3,
      },
      "start": Position {
        "column": 2,
        "line": 3,
      },
    },
    "name": "./b",
    "source": "./b",
  },
]
`;

exports[`import visitors imports/importAlias.js 1`] = `
Array [
  Object {
    "alias": "b",
    "loc": SourceLocation {
      "end": Position {
        "column": 15,
        "line": 1,
      },
      "start": Position {
        "column": 9,
        "line": 1,
      },
    },
    "name": "a",
    "source": "./base",
  },
]
`;

exports[`import visitors imports/importAll.js 1`] = `
Array [
  Object {
    "alias": "base",
    "loc": SourceLocation {
      "end": Position {
        "column": 16,
        "line": 1,
      },
      "start": Position {
        "column": 7,
        "line": 1,
      },
    },
    "name": "*",
    "source": "./base",
  },
]
`;

exports[`import visitors imports/importCss.js 1`] = `
Array [
  Object {
    "alias": "styles",
    "loc": SourceLocation {
      "end": Position {
        "column": 13,
        "line": 1,
      },
      "start": Position {
        "column": 7,
        "line": 1,
      },
    },
    "name": "default",
    "source": "../nonjs/style.css",
  },
]
`;

exports[`import visitors imports/importDefault.js 1`] = `
Array [
  Object {
    "alias": "base",
    "loc": SourceLocation {
      "end": Position {
        "column": 11,
        "line": 1,
      },
      "start": Position {
        "column": 7,
        "line": 1,
      },
    },
    "name": "default",
    "source": "./base",
  },
]
`;

exports[`import visitors imports/importNamed.js 1`] = `
Array [
  Object {
    "alias": "a",
    "loc": SourceLocation {
      "end": Position {
        "column": 10,
        "line": 1,
      },
      "start": Position {
        "column": 9,
        "line": 1,
      },
    },
    "name": "a",
    "source": "./base",
  },
]
`;

exports[`import visitors imports/noImports.js 1`] = `Array []`;
